### ISSUE TITLE
Array handlers are skipped when @marko/run is used with eBay/arc

### ISSUE THREAD

**Issue by vwong** [marko-js/run#72]
I have a handler `+handler.ts` and a variant of the handler `+handler[variant].ts`, both of which contains array handlers. Eg,

```
export const GET: MarkoRun.Handler[] = [
  () => { ... },
  () => { ... }
]
```

When I run this in *production* mode, the handlers are not executed.

I traced it down, and patch my local version with the following:

```
diff --git a/node_modules/@marko/run/dist/runtime/internal.js b/node_modules/@marko/run/dist/runtime/internal.js
index 2f4b874..a95c96f 100644
--- a/node_modules/@marko/run/dist/runtime/internal.js
+++ b/node_modules/@marko/run/dist/runtime/internal.js
@@ -106,7 +106,7 @@ function compose(handlers) {
 function normalize(obj) {
   if (typeof obj === "function") {
     return obj;
-  } else if (Array.isArray(obj)) {
+  } else if (Array.isArray(obj) || obj.length) {
     return compose(obj);
   } else if (obj instanceof Promise) {
     const promise = obj.then((value) => {
```

When used with Arc, `obj` is a `Proxy` for the array and so will fail the `Array.isArray` branch. Adding the 2nd condition works, at least for my code-base. Not sure if there is a more elegant solution.

**Comment by rturnq** [comment]
Thanks for the report and investigating the cause. I've made a [PR to Arc](https://github.com/eBay/arc/pull/41) which will preserve arrays when they are proxied so the Array.isArray check will work.

**Comment by DylanPiercey** [comment]
Fixed in `arc-server@3.1.1`
